src/main/java/com/digitalpebble/storm/crawler/bolt/ParserBolt.java

```diff
@@ -300,8 +305,17 @@ public void execute(Tuple tuple) {
             }
         }
 
-        collector.emit(tuple, new Values(url, content, metadata, text.trim(),
-                slinks));
+        for (String outlink : slinks) {
+            // configure which metadata gets inherited from parent
+            Map<String, String[]> linkMetadata = metadataTransfer
+                    .getMetaForOutlink(metadata);
+            collector
+                    .emit(com.digitalpebble.storm.crawler.Constants.StatusStreamName,
+                            tuple, new Values(outlink, linkMetadata,
```

Comment:
You misunderstood me. When emitting the outlinks, how about emitting within that tuple, the page URL as well. So your code would be 
```
           collector
                   .emit(com.digitalpebble.storm.crawler.Constants.StatusStreamName,
                           tuple, new Values(url, outlink, linkMetadata,
                                   Status.DISCOVERED));
```